src/chocolatey/infrastructure.app/nuget/NuGetEndpointResources.cs

```diff
@@ -105,21 +107,21 @@ public PackageSearchResource SearchResource
 
         public SourceRepository Source { get; private set; }
 
-        public static NuGetEndpointResources GetResourcesBySource(SourceRepository source)
+        public static NuGetEndpointResources GetResourcesBySource(SourceRepository source, ChocolateySourceCacheContext cacheContext)
         {
             return _cachedResources.GetOrAdd(source, (key) =>
             {
-                var endpointResource = new NuGetEndpointResources(key);
+                var endpointResource = new NuGetEndpointResources(key, cacheContext);
 
                 return endpointResource;
             });
         }
 
-        public static IEnumerable<NuGetEndpointResources> GetResourcesBySource(IEnumerable<SourceRepository> sources)
+        public static IEnumerable<NuGetEndpointResources> GetResourcesBySource(IEnumerable<SourceRepository> sources, ChocolateySourceCacheContext cacheContext)

```

Comment:
This was missed initially, we need to add this as an overload and make the old method obsolete.